woocommerce/woocommerce-gutenberg-products-block
# Update Filter Products by Price block to work with PHP templates

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6135
Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6139

Filter Products by Price block will only work with the PHP rendered `Classic Template` block if the filters are applied to the URL and the page is re-rendered. This behaviour happens with the Classic Themes such as Storefront.

**Changes introduced:**
* Use `asset_data_registry` to pass data to the frontend when the PHP template is being used. This will let the JS components know how to behave when interacting with them as opposed to the All Products block.
* Refactor the Filter Products by Price block to get filter params, append them to the URL and reload the page.

### Manual Testing

1. With a block theme, go to Appearance > Editor > Template > Product Catalog.
2. Add the Filter Products by Price block above the product grid and set _Filter Button_ attribute to true.
3. Change the values of the price slider.
4. Notice the page reloads and the list of products updates accordingly.

![Kooha-03-28-2022-16-01-09b](https://user-images.githubusercontent.com/3616980/160415342-25b2e1a1-4c48-4837-9dcc-b4d7a71b2886.gif)

### Changelog

> Allow adding the Filter Products by Price block to Product Catalog templates to filter products.
